suite/volume/volume_setup.go

```diff
@@ -0,0 +1,335 @@
+	log.Debug().Interface("contracts_to_jobs", contractToJobsMap).Msg("debug data for per round metrics")
+	return &FluxTest{
+		Test: Test{
+			DefaultSetup:            s,
+			OnChainCheckAttemptsOpt: spec.OnChainCheckAttemptsOpt,
+			Nodes:                   clNodes,
+			Adapter:                 adapter,
+			Prom:                    prom,
+			EthLog:                  ethlog.NewEthLog(s.Client.(*client.EthereumClient).Client, zerolog.InfoLevel),
+		},
+		FluxInstances:      &fluxInstances,
+		ContractsToJobsMap: contractToJobsMap,
+		NodesByHostPort:    nodesByHostPort,
+		roundsDurationData: make([]float64, 0),
+		AlreadySeenRuns:    make(map[string]bool),
+	}
+}
+
+// deployFluxInstance deploy one flux instance concurrently, add jobs to all the nodes
+func deployFluxInstance(d *FluxInstanceDeployment) {
+	go func() {
+		defer d.Wg.Done()
+		log.Info().Int("instance_id", d.Index).Msg("deploying contracts instance")
+		fluxInstance, err := d.Suite.Deployer.DeployFluxAggregatorContract(d.Suite.Wallets.Default(), d.Spec.FluxOptions)
+		Expect(err).ShouldNot(HaveOccurred())
+		err = fluxInstance.Fund(d.Suite.Wallets.Default(), big.NewInt(0), big.NewInt(1e18))
+		Expect(err).ShouldNot(HaveOccurred())
+		err = fluxInstance.UpdateAvailableFunds(context.Background(), d.Suite.Wallets.Default())
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// set oracles and submissions
+		err = fluxInstance.SetOracles(d.Suite.Wallets.Default(),
+			contracts.SetOraclesOptions{
+				AddList:            d.Oracles,
+				RemoveList:         []common.Address{},
+				AdminList:          d.Oracles,
+				MinSubmissions:     uint32(d.Spec.RequiredSubmissions),
+				MaxSubmissions:     uint32(d.Spec.RequiredSubmissions),
+				RestartDelayRounds: uint32(d.Spec.RestartDelayRounds),
+			})
+		Expect(err).ShouldNot(HaveOccurred())
+		for _, n := range d.Nodes {
+			fluxSpec := &client.FluxMonitorJobSpec{
+				Name:            fmt.Sprintf("%s_%d", d.Spec.JobPrefix, d.Index),
+				ContractAddress: fluxInstance.Address(),
+				PollTimerPeriod: d.Spec.NodePollTimePeriod,
+				// it's crucial not to skew rounds schedule for that particular volume test
+				IdleTimerDisabled: true,
+				PollTimerDisabled: false,
+				ObservationSource: client.ObservationSourceSpec(d.Adapter.InsideDockerAddr + "/variable"),
+			}
+			job, err := n.CreateJob(fluxSpec)
+			Expect(err).ShouldNot(HaveOccurred())
+			d.FluxInstancesMu.Lock()
+			d.NodesByHostPort[n.URL()] = n
+			d.ContractToJobsMap[fluxInstance.Address()] = append(d.ContractToJobsMap[fluxInstance.Address()],
+				JobByInstance{
+					ID:       job.Data.ID,
+					Instance: n.URL(),
+				})
+			d.FluxInstancesMu.Unlock()
+		}
+		d.FluxInstancesMu.Lock()
+		*d.FluxInstances = append(*d.FluxInstances, fluxInstance)
+		d.FluxInstancesMu.Unlock()
+	}()
+}
+
+// runInstanceKey to record start timestamp of every run + instance
+func (vt *FluxTest) runInstanceKey(inst string, runID string) string {
+	return fmt.Sprintf("%s:%s", inst, runID)
```

Comment:
I don't understand why this is needed

Reply:
Because run ID's are not unique across nodes, so I cache them all in one map. Tried to use /v2/bulk_delete_runs but it doesn't work.